### DISCUSSION TITLE
How to re-initlaize a custom accelerator inside the training loop ?

### DISCUSSION THREAD

**Discussion by shamanez** [Lightning-AI/pytorch-lightning#5884] [DDP / multi-GPU / multi-node]
In the code I use, there is a custom accelerator (DDPAccelerator) that loads an embedding similarity search model as a separate process. Now I need to re-initialize this similar method after certain training steps. In order to do this, I need to get access to the accelerator callback mentioned in [this line](https://github.com/huggingface/transformers/blob/master/examples/research_projects/rag/finetune_rag.py#L84). 

Is this possible?

@williamFalcon

**Comment by carmocca** [comment]
The link you provided isn't working. Can you share the correct link?

**Comment by shamanez** [comment]
I have updated the link.

**Comment by carmocca** [comment]
Thanks. I didn't dive deep into the code but looks to me like you actually want use a `Plugin` (https://pytorch-lightning.readthedocs.io/en/stable/plugins.html) not an `Accelerator`. You could subclass the `DDPPlugin` to properly initialize your embeddings. That could be combined with a Callback to accomplish whatever logic you need during training using whatever state was set by the plugin.

Not sure if this helps. Would probably need more details.
